pkg/smart_wallet/init_code.go

```diff
@@ -1,92 +1,47 @@
 package smart_wallet
 
 import (
-	"context"
-	"errors"
 	"fmt"
 	"log/slog"
 
-	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/shopspring/decimal"
 )
 
-var (
-	res, ok                  = entryPointABI.Errors["SenderAddressResult"]
-	senderAddressResultError = boolMust(res, ok)
-)
-
-func IsAccountDeployed(ctx context.Context, provider ethereum.ChainStateReader, swAddress common.Address) (bool, error) {
-	byteCode, err := provider.CodeAt(ctx, swAddress, nil)
-	if err != nil {
-		return false, fmt.Errorf("failed to check if smart account is already deployed: %w", err)
-	}
-
-	// assume that the smart account is deployed if it has non-zero byte code
-	return len(byteCode) != 0, nil
-}
+func GetInitCode(smartWalletConfig Config, ownerAddress common.Address, index decimal.Decimal) ([]byte, error) {
```

Comment:
I suggest adding godoc for this function since other public functions in this file have it.

Reply:
6982100528d7e2cfb6f19798fbceeaf3e24497b1